catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/converter/mgmp/MgmpConverter.java

```diff
@@ -0,0 +1,914 @@
+                                    northBoundLatitude.getAsDouble());
+
+                            geographicElementIndex++;
+                        }
+
+                    } catch (ParseException e) {
+                        LOGGER.debug("unable to set location: wkt={}", location, e);
+                    }
+                });
+
+    }
+
+    private void addBoundingBoxElement(String path, Double value) {
+        pathValueTracker.add(new Path(replaceIndex(path, geographicElementIndex)),
+                String.format(BOUNDING_BOX_FORMAT, value));
+    }
+
+    private void addGeographicIdentifier() {
+        addMultiValues(getValues(Location.COUNTRY_CODE),
+                MgmpConstants.GMD_COUNTRY_CODE_PATH,
+                geographicElementIndex);
+        addMultiValues(createListOfStrings(getValues(Location.COUNTRY_CODE), "ISO3166-1-a3"),
+                MgmpConstants.GMD_COUNTRY_CODE_SPACE_PATH,
+                geographicElementIndex);
+        geographicElementIndex += getValues(Location.COUNTRY_CODE).size();
+    }
+
+    private void addMdIdentificationStatus() {
+        addFieldIfString(GmdConstants.RESOURCE_STATUS,
+                GmdConstants.RESOURCE_STATUS_PATH,
+                () -> pathValueTracker.add(new Path(MgmpConstants.RESOURCE_STATUS_CODE_LIST_PATH),
+                        MgmpConstants.MGMP_PROGRESS_CODE));
+    }
+
+    private String generateGmlGuid() {
+        return "GMLID_" + UUID.randomUUID()
+                .toString();
+    }
+
+    private void addTemporalElements() {
+
+        List<String> starts = getDateStrings(DateTime.START);
+        List<String> ends = getDateStrings(DateTime.END);
+
+        if (starts.size() == ends.size()) {
+            for (int i = 0; i < starts.size(); i++) {
+
+                String start = starts.get(i);
+                String end = ends.get(i);
+
+                int elementIndex = i + 1;
+
+                if (start.equals(end)) {
+                    pathValueTracker.add(new Path(replaceIndex(MgmpConstants.GMD_TEMPORAL_TIME_INSTANT_ID_PATH,
+                            elementIndex)), gmlIdSupplier.get());
+                    pathValueTracker.add(new Path(replaceIndex(MgmpConstants.GMD_TEMPORAL_INSTANT_PATH,
+                            elementIndex)), start);
+                } else {
+                    pathValueTracker.add(new Path(replaceIndex(MgmpConstants.GMD_TEMPORAL_TIME_PERIOD_ID_PATH,
+                            elementIndex)), gmlIdSupplier.get());
+                    pathValueTracker.add(new Path(replaceIndex(MgmpConstants.GMD_TEMPORAL_START_PATH,
+                            elementIndex)), start);
+                    pathValueTracker.add(new Path(replaceIndex(MgmpConstants.GMD_TEMPORAL_END_PATH,
+                            elementIndex)), end);
+                }
+            }
+        }
+    }
+
+    private void addVerticalElement() {
+
+        final AtomicBoolean addVerticalCRS = new AtomicBoolean(false);
```

Comment:
I understand why we're using the `AtomicBoolean` here, but it's a little misleading since nothing here needs to be thread safe. Should we use `final boolean[] addVerticalCrs = new boolean { false }` or `final Holder<Boolean> addVerticalCrs = new Holder<>(false);` instead?